### PR TITLE
feat: added support for icomoon icons

### PR DESCRIPTION
**Pull request by ayadav-eightfold**
Added support for icomoon

### CODE REVIEW

File: src/components/Icomoon/Icon.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        <span
+            data-test-id={dataTestId}
+            aria-hidden={ariaHidden}
+            className={iconClassNames}
+            id={id}
+            role={role}
+        >
+            <IconSetContext.Consumer>
+                {(iconSet) => (
+                    <IcomoonReact
```
Comment (ayadav-eightfold) [inline src/components/Icomoon/Icon.tsx:37]: Add aria and role attributes

File: src/components/Icomoon/Icon.types.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    Medium = '20px',
+    Small = '16px',
+    XSmall = '14px',
+}
+
+export interface IcomoonProps extends OcBaseProps<HTMLSpanElement> {
+    /**
+     * The icon is aria-hidden.
+     * @default false
+     */
```
Comment (ayadav-eightfold) [inline src/components/Icomoon/Icon.types.ts:14]: update props

File: src/components/Icomoon/Icon.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import { Icomoon, IcomoonSize, IcomoonProvider } from './index';
+import iconSet from './selection.json';
+
+export default {
+    title: 'Icomoon',
+    component: Icomoon,
+};
+
+export const Icons = () => (
+    <IcomoonProvider value={iconSet}>
```
Comment (ychhabra-eightfold) [inline src/components/Icomoon/Icon.stories.tsx:11]: Let move this to the config provider

File: src/components/Icomoon/Icon.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    return (
+        <span
+            data-test-id={dataTestId}
+            aria-hidden={ariaHidden}
+            className={iconClassNames}
+            id={id}
+            role={role}
+        >
+            <IconSetContext.Consumer>
```
Comment (ychhabra-eightfold) [inline src/components/Icomoon/Icon.tsx:35]: Can we move this to this icon component based on a prop return material icon or icomooon icon ? So one component can serve both purposes.

File: src/components/Icomoon/selection.json [hunk 5]
```diff
@@ -0,0 +1,183 @@
+{
+    "IcoMoonType": "selection",
```
Comment (ychhabra-eightfold) [inline src/components/Icomoon/selection.json:2]: i hope this is a sample just for the storybook

File: package.json [hunk 6]
```diff
@@ -125,6 +125,7 @@
         "html-replace-webpack-plugin": "2.5.6",
         "html-webpack-plugin": "5.5.0",
         "husky": "7.0.4",
+        "icomoon-react": "^3.0.0",
```
Comment (ychhabra-eightfold) [inline package.json:128]:
```suggestion
        "icomoon-react": "3.0.0",
```

**Review by ychhabra-eightfold — CHANGES_REQUESTED** [review]

File: src/components/Icomoon/Icon.tsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
+import React, { FC, useContext } from 'react';
+import { mergeClasses } from '../../shared/utilities';
+import { IcomoonProps, IcomoonSize } from './';
+import IcomoonReact from 'icomoon-react';
+
+import styles from './icon.module.scss';
+
+const IconSetContext = React.createContext<object | null>(null);
```
Comment (ayadav-eightfold) [inline src/components/Icomoon/Icon.tsx:8]: Merge with Icon Component

**Comment by ychhabra-eightfold** [comment]
Closing pr as this duplicates https://github.com/EightfoldAI/octuple/pull/129

**Pull request closed without merging**